src/compiler/checker.ts

```diff
@@ -16239,6 +16239,7 @@ namespace ts {
                 if (unmatchedProperty.valueDeclaration
                     && isNamedDeclaration(unmatchedProperty.valueDeclaration)
                     && isPrivateIdentifier(unmatchedProperty.valueDeclaration.name)
+                    && source.symbol

```

Comment:
`source` uses the type `Type`. Based on the `Type` definition property `symbol` cannot be `undefined`.
https://github.com/microsoft/TypeScript/blob/56b6d0d6663b71588dea2cb5ccdd88ba904b0795/src/compiler/types.ts#L4365

Are there are other cases when `symbol` can be `undefined`? Maybe make sense to change type definition from `symbol: Symbol;` to `symbol: Symbol | undefined` in order to avoid such bugs?

What do you @DanielRosenwasser @RyanCavanaugh @weswigham think?